### ISSUE TITLE
Option to have a private/stealth hidden service

### ISSUE THREAD

**Issue by noplanman** [pparent76/Own-Mailbox_debian_install#11]
During the installation, it would be cool if power-users could choose to have a private/stealth hidden service.

Or if not in the main installation wizard, maybe add a bash script that helps set it up so that users could SSH into their Omb and activate it.

See the specific Tor features from the manual:
https://www.torproject.org/docs/tor-manual.html.en
HiddenServiceAuthorizeClient
HidServAuth

**Comment by pparent76** [comment]
Hum,

I'm not sure how it could be helpfull to have a private hidden service, since the principale of email over tor is that the tor hidden service must be reachable by other email servers.

If the hidden service is private then nothing will work any-more.

Or do you mean a secondary private hidden service for particular things (like remote ssh)?

Pierre.

**Comment by noplanman** [comment]
Yes you're right.

The email servers communicate through the proxy, right? Or do they send the mails directly to the hidden service?
I thought that the mail server (in this case the proxy, like omb.one) handles the delivery of the mail to the hidden service.
If I understood that correctly, the proxy could save the secret authorization code to have access, and the user/client would have it to access the webmail.

Have I understood the system correctly?

**Comment by pparent76** [comment]
Well if possible they communicate directly through the hidden service. For example an Own-Mailbox reaches another one directly through its tor hidden service. This allows meta-data protection: the information that 2 given .omb.one email addresses exchanged emails in the first place is protected from everybody but the 2 Own-Mailboxes.

**Comment by noplanman** [comment]
How do the different Own-Mailboxes know about each other?
Is there any diagram of flow chart of what happens with an email that gets sent to email address of the proxy? e.g. armando@noplanman.omb.one
For it to reach its destination it inevitably goes through the proxy, right? Or am I getting all confused now... 😕

**Comment by pparent76** [comment]
No an email from pierre@parent.omb.one to armando@noplanman.omb.one will not go through the proxy. (Also it is easy for someone who knows how to do it, to send an email to armando@noplanman.omb.one from any Gnu/Linux machine without going through the proxy, and while protecting meta-data. Probably this should be made easily accessible, through a Debian package or something, later on).

In order to understand how this works, and how Own-Mailboxes learn about each other, all the info is available in this document:
https://www.own-mailbox.com/Dns-emails-tor.pdf

Basically they learn about each other tor hidden services through DNS, and then reach each other through their tor hidden service directly.

It would probably be a good idea to edit summary diagram in the future (When I will have some time).

**Comment by noplanman** [comment]
So if I understood correctly, according to your document this is just a proposal and doesn't exist yet?
Checking the DNS records for the *.omb.one domains, there are no TXT entries, so I assume it's not active at this time.

I think I need to dig through the code a bit more to put all the puzzle pieces together and understand the data flow.

**Comment by pparent76** [comment]
It depends on what you call "exist". We cannot say for now that it exists a standard, but It exists within the Alpha release of Own-Mailbox. If in the future others decide to implement it, it will come closer to become a standard for email over tor. But whether some other people decide to implement it or not does not depend on me. This paper was greatly inspired by a paper that a developer of tor had written on the same subject, but that had become obsolete.

> Checking the DNS records for the *.omb.one domains, there are no TXT entries, so I assume it's not >active at this time.

There are! try:  dig _tormx.noplanman.omb.one TXT

**Comment by noplanman** [comment]
Riiiight, now I see, forgot to add the `_tormx.` subdomain name.

Well, it's up to you / us to make it a default now 😉👍

**Comment by pparent76** [comment]
On Own-Mailboxes it's already a default. I.e if you have a server example.org, and you create a valid _tormx.example.org record, Own-Mailboxes, will deliver emails to example.org through tor.
